scrnaseq.Rmd

```diff
@@ -221,112 +271,226 @@ for (i in seq(nrow(datasets))) {
+  return(s)
+  })
+
+# Combine cell metadata of the Seurat objects into one big metadata
+sc_cell_metadata = purrr::map_dfr(sc, function(s){ s[[]] }) %>% as.data.frame()
+sc_cell_metadata$orig.ident = factor(sc_cell_metadata$orig.ident,levels=unique(sc_cell_metadata$orig.ident))
+
+## FEATURES (only RNA assay at the moment) ##
+sc = purrr::map(list_names(sc), function(n) {
+  num_cells_expr = Matrix::rowSums(Seurat::GetAssayData(sc[[n]], slot="counts", assay="RNA") >= 1)
+  num_cells_expr_threshold = Matrix::rowSums(Seurat::GetAssayData(sc[[n]], slot="counts", assay="RNA") >= param$features_filter[[n]][["min_counts"]])
+  sc[[n]][["RNA"]] = Seurat::AddMetaData(sc[[n]][["RNA"]], data.frame(num_cells_expr,num_cells_expr_threshold))
+  return(sc[[n]])
+})
+
+
+knitr::kable(head(sc[[1]][[]], 5), align="l", caption="Cell meta-data, top 5 rows") %>%
+  kableExtra::kable_styling(bootstrap_options = c("striped", "hover"))
+
+
+knitr::kable(head(sc[[1]][["RNA"]][[]], 5), align="l", caption="Feature meta-data for assay 'RNA', top 5 rows (only first dataset shown)") %>% 
+  kableExtra::kable_styling(bootstrap_options = c("striped", "hover"))
 
-```{r qc_filter}
-# Filter cutoffs
-cells_filter = c(nFeature_RNA=200, nCount_RNA=NA, percent.mt=20)
-features_filter = c(min_counts=1, min_cells=3) # feature has to be found by at least one count in one cell 
 ```
 
-We keep cells with more than `r cells_filter["nFeature_RNA"]` detected features, and cells with less than `r cells_filter["percent.mt"]`% of mitochondrial reads. We keep features detected with at least `r features_filter["min_counts"]` count(s) in at least `r features_filter["min_cells"]` cell(s). 
 ```{r qc_plot}
-# Plot QC metrics
-features = c("nFeature_RNA", "nCount_RNA", "percent.mt")
-p = VlnPlot(sc, features=features, pt.size=0, col=param$col_samples, combine=FALSE) 
-names(p) = features
-for (i in features) {
-  p[[i]] = PlotMystyle(p[[i]], title=i, legend_position="none") + xlab("")
-  if (!is.na(cells_filter[i])) p[[i]] = p[[i]] + geom_hline(yintercept=cells_filter[i], lty=2, col="darkgrey") 
+## CELLS ##
+
+# Plot QC metrics for cells
+cell_qc_features = c("nFeature_RNA", "nCount_RNA", "percent_mt")
+if ("percent_ercc" %in% colnames(sc_cell_metadata)) cell_qc_features = c(cell_qc_features,"percent_ercc")
+cell_qc_features = values_to_names(cell_qc_features)
+
+p_list = list()
+for (i in names(cell_qc_features)) {
+  p_list[[i]]= ggplot(sc_cell_metadata,aes_string(x="orig.ident", y=i, fill="orig.ident")) +
+    geom_violin(scale="width")
+  p_list[[i]] = PlotMystyle(p_list[[i]], title=i, legend_position="none", fill=param$col_samples) + xlab("")
+  
+  # Creates a table with min/max values for filter i for each dataset
+  #
+  # Iterate over datasets
+  cell_filter_for_plot = purrr::map_dfr(names(param$cell_filter), function(n) {
+    # If filter i in cell filter of the dataset, then create dataframe with columns orig.ident, threshold and value
+    if (i %in% names(param$cell_filter[[n]])){
```

Comment:
i for indices ;-) ... again, can change it later.

Reply:
Ignored.